src/TotkRstbGenerator.Core/RstbGenerator.cs

```diff
@@ -18,19 +18,29 @@ public class RstbGenerator
     private readonly string _output;
     private readonly uint _padding;
 
-    public RstbGenerator(string romfs, string? output = null, uint padding = 0)
+    public RstbGenerator(string romfs, string? sourceRstbPath = null, string? output = null, uint padding = 0)
     {
-        _romfs = romfs;
-        _output = output ?? romfs.GetRsizetableFile();
-        _padding = padding;
-
-        string path = TotkConfig.Shared.RsizetablePath;
+        // Throw if sourceRstbPath or TotkConfig.Shared.RsizetablePath does not exist.
+        string path = sourceRstbPath ?? TotkConfig.Shared.RsizetablePath;
         if (!File.Exists(path)) {
-            throw new FileNotFoundException($"""
+            if (sourceRstbPath == null)
+            {
+                throw new FileNotFoundException($"""
                 The vanilla RSTB file 'System/Resource/ResourceSizeTable.Product.{TotkConfig.Shared.Version}.rsizetable.zs' could not be found in your game dump.
                 """);
+            }
+            else
+            {
+                throw new FileNotFoundException($"""
+                Input path: {sourceRstbPath} is not a valid RSTB file.
+                """);
+            }
         }
```

Comment:
This is just my preference for code style.

```suggestion
        if (!File.Exists(path)) {
            throw new FileNotFoundException(
                sourceRstbPath is not null
                    ? $"The file '{sourceRstbPath}' could not be found."
                    : $"The vanilla RSTB file 'System/Resource/ResourceSizeTable.Product.{TotkConfig.Shared.Version}.rsizetable.zs'" +
                        $" could not be found in your game dump."
            );
        }
```

Reply:
Super inline code ftw